client/src/main/java/io/split/client/HttpSplitChangeFetcher.java

```diff
@@ -101,44 +97,32 @@ public SplitChange fetch(long since, long sinceRBS, FetchOptions options) {
                 );
             }
 
-            String body = response.body();
+            SplitChange splitChange = new SplitChange();
             if (specVersion.equals(Spec.SPEC_1_1)) {
-                return Json.fromJson(body, SplitChangesOldPayloadDto.class).toSplitChange();
+                splitChange = convertBodyToOldSpec(response.body());
+            } else {
+                if (_lastProxyCheckTimestamp != 0) {
+                    splitChange.clearCache = true;
+                    _lastProxyCheckTimestamp = 0L;
+                }
+                splitChange = Json.fromJson(response.body(), SplitChange.class);
             }
-
-            return Json.fromJson(body, SplitChange.class);
-
+            return splitChange;
         } catch (Exception e) {
             throw new IllegalStateException(String.format("Problem fetching splitChanges since %s: %s", since, e), e);
         } finally {
             _telemetryRuntimeProducer.recordSyncLatency(HTTPLatenciesEnum.SPLITS, System.currentTimeMillis() - start);
         }
     }
 
-    public Long getLastProxyCheckTimestamp() {
-        return _lastProxyCheckTimestamp;
-    }
-
-    public void setLastProxyCheckTimestamp(long lastProxyCheckTimestamp) {
-        synchronized (_lock) {
-            _lastProxyCheckTimestamp = lastProxyCheckTimestamp;
-        }
-    }
-
-    private ChangeDto<RuleBasedSegment> createEmptyDTO() {
-        ChangeDto<RuleBasedSegment> dto = new ChangeDto<>();
-        dto.d = new ArrayList<>();
-        dto.t = -1;
-        dto.s = -1;
-        return dto;
+    private SplitChange convertBodyToOldSpec(String body) {
+        return Json.fromJson(body, SplitChangesOldPayloadDto.class).toSplitChange();
     }
 
     private URI buildURL(FetchOptions options, long since, long sinceRBS) throws URISyntaxException {
         URIBuilder uriBuilder = new URIBuilder(_target).addParameter(SPEC, "" + specVersion);
         uriBuilder.addParameter(SINCE, "" + since);
-        if (specVersion.equals(SPEC_1_3)) {
-            uriBuilder.addParameter(RB_SINCE, "" + sinceRBS);
-        }
+        uriBuilder.addParameter(RB_SINCE, "" + sinceRBS);
```

Comment:
why did you remove this `if`?

Reply:
resolving conflicts artifacts, adding it back